driver/optcompile.ml

```diff
@@ -58,7 +58,13 @@ let print_if ppf flag printer arg =
 let (++) x f = f x
 let (+++) (x, y) f = (x, f y)
 
-let implementation ppf sourcefile outputprefix =
+let do_transl modulename modul =
+  let id, (lam, size) =
+    Translmod.transl_implementation_flambda modulename modul
```

Comment:
What about changing transl_implementation_flambda in order to directly return `((id, size), lam)`.  The only other place where it is called has an immediate pattern to deconstruct the result anyway.